src/scripts/main.js

```diff
@@ -1 +1,66 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  document.body.addEventListener('click', () => {
+    resolve('First promise was resolved');
+  });
+
+  // eslint-disable-next-line prefer-promise-reject-errors
+  setTimeout(() => reject('First promise was rejected'), 3000);
+});
+
+const secondPromise = new Promise((resolve, reject) => {
+  document.body.addEventListener('click', () => {
+    resolve('Second promise was resolved');
+  });
+
+  document.body.addEventListener('contextmenu', () => {
+    resolve('Second promise was resolved');
+  });
+});
+
+const thirdPromise = new Promise((resolve, reject) => {
+  let leftClick = false;
+  let rightClick = false;
+
+  document.body.addEventListener('contextmenu', () => {
+    rightClick = true;
+
+    if (leftClick && rightClick) {
+      resolve('Third promise was resolved');
+    }
+  });
+
+  document.body.addEventListener('click', () => {
+    leftClick = true;
+
+    if (leftClick && rightClick) {
+      resolve('Third promise was resolved');
+    }
+  });
```

Comment:
In the third promise, both event listeners can potentially resolve the promise multiple times if both events occur more than once. Consider adding a mechanism to ensure the promise resolves only once, such as using a flag to track if it has already been resolved.